### PR TITLE
[#12] modernize and reach parity with invenio-subjects-mesh-lite

### LINKED ISSUE #12: Create Lite version with only topical headings (no qualifiers)
**Environment:** Invenio v9.1

**Problem:** It is not possible to select MeSH terms that are both headings and subheadings, for example, Blood, Cytology, or Genetics

**Steps to Reproduce:**

1. Set up InvenioRDM and install invenio-subjects-mesh
2. Log in and start a new upload
3. In the 'Search for a subject by name' box, type 'blood' or 'cytology' or 'genetics'

**Results:**

- The autocomplete dropdown populates with MeSH headings that use the blood, cytology, or genetics subheadings. 
- The list of heading suggestions in the list begin with the letter "H"; the sort order is non-obvious.
- The length of the drop-down is limited, so the main headings for these words never appear and therefore cannot be selected and added to a record.
- These headings can only be found if you type at least four letters of an allowable subheading for the main headings; the main headings cannot be used as standalone headings.

**Expected Results:**
Main headings should appear first in the drop down, before any use as subheadings. This seems like an index weighting issue.

**Suggestions:**
Subheadings should perhaps not appear at all. Please consider releasing a version of this vocabulary that only includes main headings. There are ~30,000 MeSH headings, which is plenty for many users' needs,  and the inclusion of heading-subheading pairs blows that up to something like ~650,000 terms, which seems excessive and requires more expertise to apply correctly.

### PR DESCRIPTION
**Pull request by fenekku**
- modernize package + use galter-subjects-utils
- release: v2023.1.2
- closes #12 (really closed by https://github.com/galterlibrary/invenio-subjects-mesh-lite/pull/1 , but this puts a cap on the whole process)

### CODE REVIEW

**Review by sharpattack — APPROVED** [review]
Thought about pulling the branch, and following the instructions for a thorough review. Looks good though so I am approving. If it is not a bad idea to walk through it, let me know and I can try.

File: README.md [hunk 1]
```diff
@@ -1,59 +1,111 @@
...
+pipenv install invenio-subjects-mesh
+```
 
 This will add it to your Pipfile.
 
-### Versions
+## Versions
 
-This repository follows [calendar versioning](https://calver.org/):
+This repository follows [calendar versioning](https://calver.org/) for year and month:
```
Comment (sharpattack) [inline README.md:28]: Noticed this after reviewing another PR, but the version includes month? Should this be `2023.11.2` and not `2023.1.2`?

**Review by mattbaumann1 — APPROVED** [review]

**Review by carsonicator — APPROVED** [review]
Thanks @fenekku

File: README.md (hunk 1 above)
Comment (fenekku) [reply README.md:28]: The `2023.1.2` is actually correct - I haven't updated the vocabulary list with MeSH terms from this month. Those are still the terms from 2023-01 . The `2` is just for a patch. Thanks for the double-check though!

**Comment by fenekku** [comment]
@sharpattack it's not a bad idea to try the CLI locally when all the PRs have been merged and released. The changes are more for project management: reducing duplicate code, reducing different technologies, reducing spread of files.

**Pull request merged**